golang/cli_test/test_helpers.go

```diff
@@ -0,0 +1,445 @@
+package clitest
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/stretchr/testify/require"
+	app "github.com/summa-tx/relays/golang"
+	rtypes "github.com/summa-tx/relays/golang/x/relay/types"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	clientkeys "github.com/cosmos/cosmos-sdk/client/keys"
+	"github.com/cosmos/cosmos-sdk/crypto/keys"
+	"github.com/cosmos/cosmos-sdk/server"
+	"github.com/cosmos/cosmos-sdk/tests"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+const (
+	keyFoo     = "foo"
+	stakeDenom = "stake"
+	fooDenom   = "footoken"
+)
+
+var startCoins = sdk.NewCoins(
+	sdk.NewCoin(stakeDenom, sdk.TokensFromConsensusPower(1000000)),
+	sdk.NewCoin(fooDenom, sdk.TokensFromConsensusPower(1000000)),
+)
+
+type TestData struct {
+	GenesisHeaders []rtypes.BitcoinHeader
+	NewDiffHeaders []rtypes.BitcoinHeader
+	NewHeaders     []rtypes.BitcoinHeader
+}
+
+// Fixtures is used to setup the testing environment
+type Fixtures struct {
+	BuildDir       string
+	RootDir        string
+	RelaydBinary   string
+	RelaycliBinary string
+	ChainID        string
+	RPCAddr        string
+	Port           string
+	RelaydHome     string
+	RelaycliHome   string
+	P2PAddr        string
+	T              *testing.T
+}
+
+//////////////////////////////////////////////////////////////////////////////////////
+// Instantiation /////////////////////////////////////////////////////////////////////
+//////////////////////////////////////////////////////////////////////////////////////
+
+// NewFixtures creates a new instance of Fixtures with many vars set
+func NewFixtures(t *testing.T) *Fixtures {
+	tmpDir, err := ioutil.TempDir("", "relay_integration_"+strings.TrimPrefix(t.Name(), "TestRelay/")+"_")
+	require.NoError(t, err)
+	servAddr, port, err := server.FreeTCPAddr()
+	require.NoError(t, err)
+
+	p2pAddr, _, err := server.FreeTCPAddr()
+	require.NoError(t, err)
+
+	buildDir := os.Getenv("BUILDDIR")
+
+	if buildDir == "" {
+		buildDir, err = filepath.Abs("../build/")
+		require.NoError(t, err)
+	}
+
```

Comment:
here's a simple file exists function. Run this on `relayd` and `relaycli` and throw an error if false. make sure the error includes the path you checked for them
```
func fileExists(filename string) bool {
    info, err := os.Stat(filename)
    if os.IsNotExist(err) {
        return false
    }
    return !info.IsDir()
}
```